src/main/java/uk/gov/ons/ssdc/caseprocessor/messaging/NewCaseReceiver.java

```diff
@@ -93,44 +112,36 @@ public void receiveNewCase(Message<byte[]> message) {
           "Attempt to send sensitive data to RM which was not part of defined sample");
     }
 
+    return sensitiveColumns;
+  }
+
+  private void checkNewSampleWithinSampleDefinition(
+      ColumnValidator[] columnValidators, NewCase newCasePayload) {
+    Set<String> nonSensitiveColumns =
+        Arrays.stream(columnValidators)
+            .filter(columnValidator -> !columnValidator.isSensitive())
+            .map(ColumnValidator::getColumnName)
+            .collect(Collectors.toSet());
+    if (!nonSensitiveColumns.containsAll(newCasePayload.getSample().keySet())) {
+      throw new RuntimeException("Attempt to send data to RM which was not part of defined sample");
+    }
+  }
+
+  private void validateNewCase(NewCase newCasePayload, ColumnValidator[] columnValidators) {
     for (ColumnValidator columnValidator : columnValidators) {
       Optional<String> columnValidationErrors;
 
       if (columnValidator.isSensitive()) {
-        columnValidationErrors = columnValidator.validateRow(newCasePayload.getSampleSensitive());
+        columnValidationErrors =
+            columnValidator.validateRow(newCasePayload.getSampleSensitive(), true);
       } else {
-        columnValidationErrors = columnValidator.validateRow(newCasePayload.getSample());
+        columnValidationErrors = columnValidator.validateRow(newCasePayload.getSample(), true);
       }
 
       if (columnValidationErrors.isPresent()) {
-        throw new RuntimeException(
-            String.format(
-                "New case event failed validation on column \"%s\"",
-                columnValidator.getColumnName()));
+        throw new RuntimeException("NEW_CASE event: " + columnValidationErrors.get());
```

Comment:
might as well gather ALL the column validation errors and throw this at the end, rather than just bombing out on the first one?